src/backend/vector/packed_simd.rs

```diff
@@ -0,0 +1,252 @@
+            }
+        }
+
+        impl BitAnd for $ty {
+            type Output = Self;
+
+            #[inline]
+            fn bitand(self, rhs: $ty) -> Self {
+                unsafe { core::arch::x86_64::_mm256_and_si256(self.0, rhs.0).into_bits() }
+            }
+        }
+
+        impl BitXor for $ty {
+            type Output = Self;
+
+            #[inline]
+            fn bitxor(self, rhs: $ty) -> Self {
+                unsafe { core::arch::x86_64::_mm256_xor_si256(self.0, rhs.0).into_bits() }
+            }
+        }
+
+        impl BitAndAssign for $ty {
+            #[inline]
+            fn bitand_assign(&mut self, rhs: $ty) {
+                *self = *self & rhs;
+            }
+        }
+
+        impl BitXorAssign for $ty {
+            #[inline]
+            fn bitxor_assign(&mut self, rhs: $ty) {
+                *self = *self ^ rhs;
+            }
+        }
+
+        #[allow(dead_code)]
+        impl $ty {
+            #[inline]
+            pub fn shl<const N: i32>(self) -> Self {
+                unsafe { core::arch::x86_64::$shl_intrinsic(self.0, N).into_bits() }
+            }
+
+            #[inline]
+            pub fn shr<const N: i32>(self) -> Self {
+                unsafe { core::arch::x86_64::$shr_intrinsic(self.0, N).into_bits() }
+            }
+
+            #[inline]
+            pub fn extract<const N: i32>(self) -> $lane_ty {
+                unsafe { core::arch::x86_64::$extract_intrinsic(self.0, N) as $lane_ty }
+            }
+        }
+    };
+}
+
+macro_rules! impl_conv {
+    ($src:ident => $($dst:ident),+) => {
+        $(
+            impl IntoBits<$dst> for $src {
+                #[inline]
+                fn into_bits(self) -> $dst {
+                    $dst(self.0)
+                }
+            }
+        )+
+    }
+}
+
+impl_shared!(
+    u64x4,
+    u64,
+    _mm256_add_epi64,
+    _mm256_sub_epi64,
+    _mm256_slli_epi64,
+    _mm256_srli_epi64,
+    _mm256_extract_epi64
+);
+impl_shared!(
+    u32x8,
+    u32,
+    _mm256_add_epi32,
+    _mm256_sub_epi32,
+    _mm256_slli_epi32,
+    _mm256_srli_epi32,
+    _mm256_extract_epi32
+);
+impl_shared!(
+    i32x8,
+    i32,
+    _mm256_add_epi32,
+    _mm256_sub_epi32,
+    _mm256_slli_epi32,
+    _mm256_srli_epi32,
+    _mm256_extract_epi32
+);
+
+impl_conv!(u64x4 => u32x8, i32x8);
+
+#[allow(dead_code)]
+impl u64x4 {
+    #[inline]
+    pub const fn new_const(x0: u64, x1: u64, x2: u64, x3: u64) -> Self {
+        unsafe { Self(core::mem::transmute([x0, x1, x2, x3])) }
+    }
+
+    #[inline]
+    pub const fn splat_const<const N: u64>() -> Self {
+        Self::new_const(N, N, N, N)
+    }
+
+    #[inline]
+    pub fn new(x0: u64, x1: u64, x2: u64, x3: u64) -> Self {
+        unsafe {
+            Self(core::arch::x86_64::_mm256_set_epi64x(
+                x3 as i64, x2 as i64, x1 as i64, x0 as i64,
+            ))
+        }
+    }
+
+    #[inline]
```

Comment:
A comment here and for `splat` would be nice. "Constructs `Self` as 4 copies of the given `u64`" or something. Ditto for `splat_const`.